alg/teca_damper.cxx

```diff
@@ -0,0 +1,217 @@
+            input_data[0]);
+    if (!in_mesh)
+    {
+        TECA_ERROR("empty input, or not a mesh")
+        return nullptr;
+    }
+
+    // create output and copy metadata, coordinates, etc
+    p_teca_cartesian_mesh out_mesh =
+        std::dynamic_pointer_cast<teca_cartesian_mesh>(in_mesh->new_instance());
+
+    out_mesh->shallow_copy(
+        std::const_pointer_cast<teca_cartesian_mesh>(in_mesh));
+
+    // get the input array names
+    std::vector<std::string> damper_vars = this->get_damper_variables(request);
+    if (damper_vars.empty())
+    {
+        TECA_ERROR("A damper variable was not specified")
+        return nullptr;
+    }
+
+    // get threshold values
+    double filter_lat_width_val = this->filter_lat_width_value;
+
+    if (std::isnan(filter_lat_width_val)
+        && request.has("teca_damper::filter_lat_width_value"))
+        request.get("teca_damper::filter_lat_width_value", filter_lat_width_val);
+
+    if (std::isnan(filter_lat_width_val))
+    {
+        TECA_ERROR("A filter_lat_width (delta-Y) value was not specified")
+        return nullptr;
+    }
+
+    // get the coordinate axes
+    const_p_teca_variant_array yc = in_mesh->get_y_coordinates();
+    const_p_teca_variant_array xc = in_mesh->get_x_coordinates();
+
+    size_t n_lat = yc->size();
+    size_t n_lon = xc->size();
+
+    p_teca_variant_array filter_array = yc->new_instance(n_lat);
+
+    // Get the gaussian filter
+    NESTED_TEMPLATE_DISPATCH_FP(teca_variant_array_impl,
+        filter_array.get(),
+        _COORD,
+
+        const NT_COORD *p_yc = static_cast<const TT_COORD*>(yc.get())->get();
+
+        NT_COORD *p_filter_array = static_cast<TT_COORD*>(filter_array.get())->get();
+
+        ::get_lat_filter(p_filter_array, p_yc, n_lat, filter_lat_width_val);
+
+        size_t n_arrays = damper_vars.size();
+        for (size_t i = 0; i < n_arrays; ++i)
+        {
+            // get the input array
+            const_p_teca_variant_array input_array
+                = out_mesh->get_point_arrays()->get(damper_vars[i]);
+            if (!input_array)
+            {
+                TECA_ERROR("damper variable \"" << damper_vars[i]
+                    << "\" is not in the input")
+                return nullptr;
+            }
+
+            // apply the gaussian damper 
+            size_t n_elem = input_array->size();
+            p_teca_variant_array damped_array = input_array->new_instance(n_elem);
+
+            TEMPLATE_DISPATCH(teca_variant_array_impl,
+                damped_array.get(),
+
+                const NT *p_in = static_cast<const TT*>(input_array.get())->get();
+                NT *p_damped_array = static_cast<TT*>(damped_array.get())->get();
+
+                ::apply_damper(p_damped_array, p_in, p_filter_array, 
+                    n_lat, n_lon);
+            )
+
+            // set the damped array in the output
+            out_mesh->get_point_arrays()->set(damper_vars[i], damped_array);
```

Comment:
these need to have a different name, otherwise they will replace the original

Reply:
Are there examples for a post-fix being added to follow the conventional way of doing it? I only found this in `teca_table_remove_rows.cxx` but I don't think it's applicable in our case:

```
void teca_table_remove_rows::set_mask_expression(const std::string &expr)
{
    if (expr == this->mask_expression)
        return;

    // convert the mask_expression to postfix, doing so here let's
    // us know which variables will be needed
    std::set<std::string> dep_vars;
    char *pfix_expr = teca_parser::infix_to_postfix(expr.c_str(), &dep_vars);
    if (!pfix_expr)
    {
        TECA_ERROR("failed to convert \"" << expr << "\" to postfix")
        return;
    }

    this->mask_expression = expr;
    this->postfix_expression = pfix_expr;
    this->dependent_variables = std::move(dep_vars);
    this->set_modified();

    free(pfix_expr);
}
```

Reply:
here's an example:
```C++
std::string var_name = "foo";
std::string post_fix = "_damped";
std::string ovar_name = var_name + post_fix;
```
if that were to run, ovar_name will contain "foo_damped"